src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java

```diff
@@ -155,26 +141,80 @@ public void testExceptionShouldBeCatchedConditionIsMetAndDefaultTimeoutIsNotOver
                     "Conditional should be true", Collections.singleton(IllegalArgumentException.class));
         } catch (org.openqa.selenium.TimeoutException e){
             DurationSample durationSample = new DurationSample(timer.duration(), waitForTimeoutCondition, defaultDeviation);
-            BrowserManager.getBrowser().quit();
-
             assertTrue(durationSample.isDurationBetweenLimits(), durationSample.toString());
+        } finally {
+            BrowserManager.getBrowser().quit();
         }
     }
 
     @Test
-    public void testTimeoutExceptionShouldNotBeThrownIfDriverConditionIsMetAndTimeoutIsNotOver(){
+    public void testTimeoutExceptionShouldNotBeThrownIfDriverConditionIsMetAndTimeoutIsNotOver() {
         Timer timer = new Timer();
+        try {
+            boolean conditionResult = ConditionalWait.waitFor((driver) ->
+                    {
+                        timer.start();
+                        return true;
+                    }, waitForTimeoutCondition, waitForTimeoutPolling,
+                    "Conditional should be true", Collections.singleton(IllegalArgumentException.class));
+            DurationSample durationSample = new DurationSample(timer.duration(), waitForTimeoutCondition);
+            assertTrue(durationSample.getDuration() < waitForTimeoutCondition);
+            assertTrue(conditionResult, "Condition result should be true");
+        } finally {
+            BrowserManager.getBrowser().quit();
+        }
+    }
 
-        ConditionalWait.waitFor((driver) ->
-                {
-                    timer.start();
-                    return true;
-                }, waitForTimeoutCondition, waitForTimeoutPolling,
-                "Conditional should be true", Collections.singleton(IllegalArgumentException.class));
+    @Test
+    public void testTrueShouldNotBeReturnedIfConditionIsMetAndTimeoutIsNotOver(){
+        Timer timer = new Timer();
+        boolean conditionResult = ConditionalWait.waitFor(() ->
+        {
+            timer.start();
+            return true;
+        }, waitForTimeoutCondition, waitForTimeoutPolling, "Timeout exception should not be thrown");
         DurationSample durationSample = new DurationSample(timer.duration(), waitForTimeoutCondition);
-        BrowserManager.getBrowser().quit();
-
         assertTrue(durationSample.getDuration() < waitForTimeoutCondition);
+        assertTrue(conditionResult, "Condition result should be true");
+    }
+
+    @Test
+    public void testFalseShouldNotBeReturnedIfConditionIsNotMetAndTimeoutIsOver(){
```

Comment:
please, rename method name as 'testfalseshould BE RETURNED'

Reply:
fixed